File: online_store/domain/models/Order.py
from online_store.domain.models.Catalog import Catalog
class Order:
    def __init__(self, customer, payment_gateway):
        self.customer = customer
        self.payment_gateway = payment_gateway

    def generate_order_summary(self, catalog:Catalog) -> str:
        # Calculate the total price of items in the order
        total_price = self.customer.shopping_cart.calculateTotalPrice()
        if total_price is None:
            return "Shopping Cart is empty"
        # Process payment with the selected payment gateway
        if self.payment_gateway.process_payment(total_price):
            for item in self.customer.shopping_cart.items:
                catalog.remove_item(item)
            self.customer.clear_cart()
            return "Payment succesful"
        else:
            return "Payment processing failed."

File: tmpsjava/online_shop/src/domain/models/Order.java
package domain.models;

import domain.models.products.ProductInterface;

public class Order {
    CustomerInterface customer;
    PaymentOptionInterface payment_gateway;
    public Order(CustomerInterface customer, PaymentOptionInterface payment_gateway){
        this.customer = customer;
        this.payment_gateway = payment_gateway;
    }
    public String generateSummary(Catalog catalog){
        float total_price = this.customer.getShoppingCart().calculateTotalPrice();
        if (total_price == 0){
            return "Shopping Cart is empty";
        }
        if (this.payment_gateway.processPayment(total_price)){
            for(ProductInterface item : this.customer.getShoppingCart().getItems()){
                catalog.remove_item(item);
            }
            this.customer.getShoppingCart().clear();
            return "Payment succesful";
        }
        return "Payment processing failed.";
    }
}
